docs/benchmark.md

```diff
@@ -12,86 +12,202 @@ It's important to look at the units than at the percentages. In general, RxDogTa
 
 Running the benchmark on a [Pixel 3](https://store.google.com/product/pixel_3_specs) yields the following results:
 
-### Event throughput (grouped by number of events)
+### Event throughput: grouped by number of events
 
 Measures the amount of time it takes for given number of elements to pass through the stream.
 
 #### 1 item (Observable)
 | Benchmark | Time (ns) | Time (ms) | Percent Increase |
 |----------|------------|-----------|------------------|
-|RxDogTagAndroidPerf.observable1_false | 212ns | 0.000ms|
-|RxDogTagAndroidPerf.observable1_true | 13,267ns | 0.013ms | 6158.02% |
+| RxDogTagAndroidPerf.observable_simple[enabled=false,times=1,guardedDelegateEnabled=false]          |         577ns |       0.001ms |
+| RxDogTagAndroidPerf.observable_simple[enabled=true,times=1,guardedDelegateEnabled=false]           |      13,903ns |       0.014ms | 2309.53% |
+| RxDogTagAndroidPerf.observable_simple[enabled=true,times=1,guardedDelegateEnabled=true]            |      14,303ns |       0.014ms | 2378.86% |
+| RxDogTagAndroidPerf.observable_simple[enabled=false,times=1,000,guardedDelegateEnabled=false]      |      23,987ns |       0.024ms | 4057.19% |
```

Comment:
The heading says: `1 item (Observable)` but the benchmark has ones where we're sending 1000 and 1_000_000. We should filter these out